Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/SonarResolveReferencesTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Tasks;
+using SonarScanner.MSBuild.Tasks.IntegrationTests;
+using TestUtilities;
+
+namespace SonarScanner.Integration.Tasks.IntegrationTests.TargetsTests
+{
+    [TestClass]
+    public class SonarResolveReferencesTests
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        [TestCategory("IsTest")]
+        public void BuildIntegration_ResolvesReferences()
+        {
+            // Arrange
+            var rootFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSnippet = $@"
+<PropertyGroup>
+    <SonarQubeTempPath>{rootFolder}</SonarQubeTempPath>
+</PropertyGroup>";
+
+            var filePath = CreateProjectFile(projectSnippet);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.DefaultBuildTarget);
+
+            // Assert
+            result.AssertTargetSucceeded(TargetConstants.DefaultBuildTarget);
+            result.AssertTargetExecuted(TargetConstants.ResolveReferencesTarget);
+            result.AssertTargetExecuted(TargetConstants.CategoriseProjectTarget);
+
+            var sonarResolvedReferences = result.GetCapturedItemValues(TargetProperties.SonarResolvedReferences);
+            sonarResolvedReferences.Should().NotBeEmpty();
+            sonarResolvedReferences.Should().Contain(x => x.Value.Contains("mscorlib"));
+        }
+
+        private string CreateProjectFile(string projectSnippet)
+        {
+            var captureReferences = $@"
```

Comment:
I would add a small comment like `// this target captures the values we are interested to test`